instructure/canvas-android
# [Flutter Parent][MBL-14213] Display message regarding old reminders

Also moves `OldAppMigrations` to the service locator and adds missing tests.

To test:
 - Build/install the native parent app, log in, and set up some reminders
 - Build/install the flutter app using the same signing key so that the app is updated instead of uninstalled
 - If no reminders were set up in the native app, or the date for all reminders are prior to today's date, then _no_ dialog should display on the dashboard.
 - If the date for one or more reminders is today or in the future, a dialog _should_ display on the dashboard.
 - After dismissing the dialog, it should never display again.
